### DISCUSSION TITLE
jpegXL recompression of jpeg ?

### DISCUSSION THREAD

**Discussion by Shmowder** [ImageMagick/ImageMagick#6046] [General]
Hi, please excuse me if this is the wrong place for this question/request.
As I understood in IM there is no option to jpegXL recompression of jpeg. I mean that I can't make exactly a recompression of something like old jpeg photos with the same quality but with a lower file size.
Is there possible to add feature like this?
Thank you.

**Comment by dlemstra** [comment]
We won't / can't add this feature because we always decode the pixels before converting the image to another format.

**Reply by Shmowder** [reply]
I think I have explained my question in a wrong way.

From the beginning I have asked on reddit why "lossless" export (with settings of -quality 100) of .avif and .jxl has bigger file size than .jpg file, if .jpg is already compressed. (I am pretty new with all these things).
https://www.reddit.com/r/AV1/comments/10u7oy8/size_of_lossless_avif_and_jxl_is_bigger_than_jpg/
People explained how it works and then I thought something like: ok, but how or what quality settings should I use if I want to make .avid or .jxl copy of an image with smaller size (because of the better codec) but with the same quality as in sours photo (that was in .jpg). And then another person explained that .avif can't do it but Jpeg-XL - can, and I should use "lossless-transcoding feature of jpeg xl" and "cjxl/libjxl" can do it.

Perhaps I use the wrong terminology, but the point of my question is: if these "cjxl/libjxl" can .jxl from .jpg (so same quality, better codec, smaller file size) to  with the same quality and it is a "lossless-transcoding feature of jpeg xl", then is it possible to add this feature in IM ?
Thank you.

**Reply by sergeevabc** [reply]
Err, I don't understand your answer, so am feeling lost now.
How to accomplish the following conversion by means of IM alone, without CJXL?

[cjxl.exe](https://github.com/libjxl/libjxl/releases/tag/v0.8.2) [image.jpg](https://s7.krakenfiles.com/uploads/09-11-2023/OSMbbh2YVA/image.jpg) image.lossless.jxl
``` console
$ cjxl image.jpg image.lossless.jxl
JPEG XL encoder v0.8.2 954b460 [SSSE3,Unknown]
Note: Implicit-default for JPEG is lossless-transcoding.
Read JPEG image with 4244645 bytes.
Encoding [Container | JPEG, lossless transcode, effort: 7 | JPEG reconstruction data],
Compressed to 3466260 bytes including container

$ djxl image.lossless.jxl image.djxl.jpg
JPEG XL decoder v0.8.2 954b460 [SSSE3,Unknown]
Read 3466260 compressed bytes.
Reconstructed to JPEG.
4048 x 3036, 5.89 MP/s [5.89, 5.89], 2.03 MB/s [2.03, 2.03], 1 reps, 2 threads.

$ xxhsum *.jpg
c8336de1b739ed3e  image.jpg
c8336de1b739ed3e  image.djxl.jpg

$ jxlinfo image.lossless.jxl
JPEG XL file format container (ISO/IEC 18181-2)                                          
JPEG XL image, 4048x3036, (possibly) lossless, 8-bit RGB                                 
Color space: 596-byte ICC profile, CMM type: "", color space: "RGB ", rendering intent: 0
JPEG bitstream reconstruction data available                                             
Brotli-compressed Exif metadata: 27514 compressed bytes
```
